api/schema.py

```diff
@@ -0,0 +1,33 @@
+from typing import Any
+
+from drf_spectacular.openapi import AutoSchema
+from rest_framework.permissions import SAFE_METHODS
+
+from api.serializers.errors import ValidationErrorSerializer
+
+
+class CustomSchema(AutoSchema):
+    """
+    Custom schema for drf-spectacular.
+
+    Differences from default schema:
+    - Default schema uses same tag - first non-parameter path part as tag for all endpoints.
+      (e.g. /api/users/ -> tag: api and also /api/other/ -> tag: api).
+      To make it more clear, we use viewset name as additional tag.
+    - We add 400 response for all endpoints with unsafe method.
```

Comment:
```suggestion
    - We override _get_response_bodies() method to add 400 response for all endpoints with unsafe methods.
```

Also, can and endpoint have an unsafe method? Or do you mean "for all requests to endpoints with unsafe methods"?

Reply:
> for all requests to endpoints with unsafe methods

This is definitely more correct. Fixed